client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -582,29 +581,44 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
     // check if there exists request for the partition, if do just register
     val requests = changePartitionRequests.computeIfAbsent(shuffleId, rpcContextRegisterFunc)
     inBatchPartitions.computeIfAbsent(shuffleId, inBatchShuffleIdRegisterFunc)
-    requests.synchronized {
-      if (requests.containsKey(partitionId)) {
-        requests.get(partitionId).add(changePartition)
-        logTrace(s"[handleChangePartitionLocation] For $shuffleId, request for same partition" +
-          s"$partitionId-$oldEpoch exists, register context.")
-        return
+    if (requests.containsKey(partitionId)) {
```

Comment:
I'm afraid there will be concurrent issues if batch is disabled

Reply:
> I'm afraid there will be concurrent issues if batch is disabled

Yea,  even for batch mode,  also have concurrent issues.

Reply:
I will close this.